platform-emf/src/main/java/com/softicar/platform/emf/management/importing/variable/find/EmfImportVariablesFinder.java

```diff
@@ -2,27 +2,46 @@
 
 import com.softicar.platform.common.container.map.set.SetMap;
 import com.softicar.platform.emf.management.importing.variable.EmfImportVariableCoordinates;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
+/**
+ * Finds variables by a defined pattern in a matrix of {@link String} tokens.
+ * <p>
+ * The variables are case insensitive. The first occurrence of a variable in the
+ * matrix determines its display name in terms of upper and lower case letters.
+ *
+ * @author Thees Köster
+ */
 public class EmfImportVariablesFinder {
 
-	private static final String VARIABLE_PATTERN = "\\$\\$[a-z]\\$\\$";
+	private static final String VARIABLE_PATTERN = "\\$\\$[a-zA-Z0-9_]+\\$\\$";
 
 	private final List<List<String>> rows;
-	private final SetMap<String, EmfImportVariableCoordinates> variableCoordinates = new SetMap<>();
+	private final Map<String, String> variableToDisplayNameMap = new HashMap<>();
+	private final SetMap<String, EmfImportVariableCoordinates> variableDisplayNameToCoordinatesMap = new SetMap<>();
 
 	public EmfImportVariablesFinder(List<List<String>> rows) {
 
 		this.rows = rows;
 	}
 
+	/**
+	 * Finds variables and their coordinates in the given matrix of
+	 * {@link String} tokens.
```

Comment:
I think we need to add to this Javadoc block:
```
<p>
Invalid variable names will be silently ignored; they will not be added to the return value, and no exception will occur. Valid variable names match the following pattern: {@value #VARIABLE_PATTERN}
```

Reply:
added as suggested